### DISCUSSION TITLE
How to setup Mail Server behind different ports than 80/443

### DISCUSSION THREAD

**Discussion by thepra** [modoboa/modoboa#2215] [General]
After installation in the nginx config files I did put 8080(http) and 8444(https) as the listening ports both in `autoconfig.<DOMAIN>.conf` and `mail.<DOMAIN>.conf` because my home has already a main server mapped to those ports, so other servers have to be mapped on different ports to become reachable, but given this environment how can I correctly adjust the server based on this?

**Comment by tonioo** [comment]
To access Modoboa web panel, it should be enough. For autoconfig, I'm not sure it is something allowed.

**Reply by thepra** [reply]
I agree, the web panel works fine and indeed autoconfig doesn't seem to work, strangely:
![immagine](https://user-images.githubusercontent.com/7032509/114164969-3747df80-992c-11eb-9420-90f38d8d6079.png)
does `autoconfig` work only on expected `80` port?
Also is there a way to manully refresh those DNS records checks?
And also the MX record should point the main domain `mail.<main-domain>` for any added domain should point to its own `mail.<different-domain>`?

**Reply by mehdiMj-ir** [reply]
It would be ideal if we could serve this app behind a NAT and on a non-standard port, for ex 9443
<img width="1920" height="866" alt="Screenshot_20251003_133518" src="https://github.com/user-attachments/assets/4292643d-984a-4111-89fd-a8a057cb81ec" />
I attempted to set it up on port 9443 by modifying both the `Nginx` listen port and the `/srv/modoboa/instance/frontend/config.json` file. I made some progress with the SSO, but after logging in, I encountered a 400 Bad Request error.
BTW
<img width="1097" height="346" alt="Screenshot_20251003_133550" src="https://github.com/user-attachments/assets/df001b61-50be-408d-a335-b05d6cd9e121" />
<img width="1920" height="883" alt="Screenshot_20251003_133730" src="https://github.com/user-attachments/assets/4010c845-fae6-411e-b11c-91190d73a28e" />

**Comment by Mark0x01** [comment]
I have the same issue, but have managed to get it a bit further on my dev server.

Modify the ALLOWED_HOSTS to include the non-standard port  e.g. mailer.local:8443

Then run manage.py load_initial_data

Then add the host without the port as the first entry.

ALLOWED_HOSTS = [
    'mailer.local',
    'mailer.local:8443',
]

Restart uwsgi and nginx.

The frontend/config.jason is now correct:
  "API_BASE_URL": "https://mailer.local:8443/api/v2",
  "API_DOC_URL": "https://mailer.local:8443/api/schema-v2/swagger/",
  "OAUTH_AUTHORITY_URL": "https://mailer.local:8443/api/o",
  "OAUTH_CLIENT_ID": "f48451d6-5917-4rrf-b50f-e97e3bce4f72",
  "OAUTH_REDIRECT_URI": "https://mailer.local:8443/login/logged",
  "OAUTH_POST_REDIRECT_URI": "https://mailer.local:8443"

I can now access the administration page at https://mailer.local:8443

I haven't tested webmail yet, but this should get my upgraded live server manageable again.

**Comment by Mark0x01** [comment]
Webmail should work, but I'm currently chasing a password failure issue that has now become way too complex.

Symptom: Dovecot authentication always fails.
dovecot[1974]: auth: sql(freddy@testdomain.com,127.0.0.1,<MGHbWqVaYkp/AAAB>): Password mismatch
dovecot[1974]: auth: oauth2(freddy@testdomain.com,127.0.0.1,<MGHbWqVaYkp/AAAB>): oauth2 failed: Introspection failed: Provided token is not valid

This was after having resolved dovecot rejecting self signed certs.

When checking Via Admin settings General, the default of sha512crypt gives " is not a valid choice.' 
Only Plain and md5crypt are offered.

What does dovecot support?
doveadm pw -l
SHA1 SSHA512 SCRAM-SHA-256 BLF-CRYPT PLAIN HMAC-MD5 OTP SHA512 SHA DES-CRYPT CRYPT SSHA MD5-CRYPT PLAIN-MD4 PLAIN-MD5 SCRAM-SHA-1 SHA512-CRYPT CLEAR CLEARTEXT SSHA256 MD5 PBKDF2 SHA256 CRAM-MD5 PLAIN-TRUNC SHA256-CRYPT SMD5 DIGEST-MD5 LDAP-MD5 

Looks ok.

Try other encryption options:
Install python argon2 components & tested they work.

Still no joy.
